plonky/src/api.rs

```diff
@@ -226,3 +228,33 @@ pub fn generate_aggregation_verifier(
     verifier::recursive_plonk_verifier::create_verifier_contract_from_default_template(config, sol);
     Result::Ok(())
 }
+
+pub fn groth16_setup(circuit_file: &String, witness: &String, curve_type: &str) -> Result<()> {
+    let mut rng = rand::thread_rng();
+    match curve_type {
+        "bn256" => {
+            let circuit: CircomCircuit<Bn256> = CircomCircuit {
+                r1cs: reader::load_r1cs(circuit_file),
+                witness: Some(reader::load_witness_from_file::<Bn256>(witness)),
+                wire_mapping: None,
+                aux_offset: 0,
+            };
+            Groth16::circuit_specific_setup(circuit, &mut rng);
```

Comment:
The result, which may be an error, is not captured.

Reply:
I think we may need to support bls12381 in plonk too. 

So we can merge the setup of plonk and groth16.  This is not first priority, we can do it after we finish all functional tests.